core/src/main/java/org/apache/carbondata/core/util/CarbonUtil.java

```diff
@@ -841,17 +837,14 @@ public static WriterCompressModel getValueCompressionModel(
      */
     for (int i = 0; i < dataTypeSelected.length; i++) {  // always 1
       ValueEncoderMeta valueEncoderMeta = encodeMetaList.get(i);
-      maxValue[i] = valueEncoderMeta.getMaxValue();
-      minValue[i] = valueEncoderMeta.getMinValue();
-      uniqueValue[i] = valueEncoderMeta.getUniqueValue();
+      max[i] = valueEncoderMeta.getMaxValue();
+      min[i] = valueEncoderMeta.getMinValue();
+      nonExistValue[i] = valueEncoderMeta.getUniqueValue();
       decimal[i] = valueEncoderMeta.getDecimal();
       type[i] = valueEncoderMeta.getType();
       dataTypeSelected[i] = valueEncoderMeta.getDataTypeSelected();
     }
-    MeasureMetaDataModel measureMetadataModel =
-        new MeasureMetaDataModel(minValue, maxValue, decimal, dataTypeSelected.length, uniqueValue,
-            type, dataTypeSelected);
-    return ValueCompressionUtil.getWriterCompressModel(measureMetadataModel);
+    return MeasurePageStatistics.build(min, max, nonExistValue, decimal, type, dataTypeSelected);
```

Comment:
Better pass `ValueEncoderMeta` list to `build` method to avoid duplicate code in other places